src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -815,6 +817,9 @@ bool CachedOnDiskReadBufferFromFile::nextImplStep()
                 }
             }
 
+            if (use_external_buffer && !internal_buffer.empty())
+                internal_buffer.resize(original_buffer_size);
```

Comment:
I do not understand this change, could you please explain why this is needed?

Reply:
we did resize on l.852 and after reading is done we need to roll it back